### PR TITLE
GEOPY-1187: Test failing on geoh5py develop branch

### PR DESCRIPTION
**Pull request by benk-mira**
**GEOPY-1187 - Test failing on geoh5py develop branch**

### CODE REVIEW

File: geoh5py/shared/utils.py [hunk 1]
```diff
@@ -262,14 +262,18 @@ def are_objects_similar(obj1, obj2, ignore):
     return attributes1 == attributes2
 
 
-def compare_entities(
+def compare_entities(  # pylint: disable=too-many-branches
```
Comment (sebhmg) [inline geoh5py/shared/utils.py:265]: disabling pylint here and there defeats the purpose of using to help maintaining a clean code base.
Current setting is already at 13 branches max...
That is a hint that this function needs to be split

**Review by sebhmg — CHANGES_REQUESTED** [review]
pylint gives an hint  that the function needs to be refactored

File: geoh5py/shared/utils.py (hunk 1 above)
Comment (benk-mira) [reply geoh5py/shared/utils.py]: Ok, I split the function up into a bunch of compare_x methods

File: geoh5py/shared/utils.py [hunk 2]
```diff
@@ -262,57 +262,70 @@ def are_objects_similar(obj1, obj2, ignore):
...
+
+
+def compare_bytes(object_a, object_b):
+    assert (
+        object_a == object_b
+    ), f"{type(object_a)} objects: {object_a}, {object_b} are not equal."
+
+
 def compare_entities(
     object_a, object_b, ignore: list | None = None, decimal: int = 6
```
Comment (sebhmg) [inline geoh5py/shared/utils.py:308]:
```suggestion
    object_a, object_b, ignore: list[str] | None = None, decimal: int = 6
```

File: geoh5py/shared/utils.py [hunk 3]
```diff
@@ -262,57 +262,70 @@ def are_objects_similar(obj1, obj2, ignore):
...
+def compare_floats(object_a, object_b, attribute, decimal: int = 6):
+    np.testing.assert_almost_equal(
+        getattr(object_a, attribute),
+        getattr(object_b, attribute),
+        decimal=decimal,
+        err_msg=f"Error comparing attribute '{attribute}'.",
+    )
+
+
+def compare_list(object_a, object_b, attribute, ignore):
```
Comment (sebhmg) [inline geoh5py/shared/utils.py:292]:
```suggestion
def compare_list(object_a, object_b, attribute: str, ignore: list[str]):
```

File: geoh5py/shared/utils.py [hunk 4]
```diff
@@ -262,57 +262,70 @@ def are_objects_similar(obj1, obj2, ignore):
...
+    else:
+        np.testing.assert_array_almost_equal(
+            attr_a,
+            getattr(object_b, attribute).tolist(),
+            decimal=decimal,
+            err_msg=f"Error comparing attribute '{attribute}'.",
+        )
+
+
+def compare_floats(object_a, object_b, attribute, decimal: int = 6):
```
Comment (sebhmg) [inline geoh5py/shared/utils.py:283]:
```suggestion
def compare_floats(object_a, object_b, attribute: str, decimal: int = 6):
```

File: geoh5py/shared/utils.py [hunk 5]
```diff
@@ -262,57 +262,70 @@ def are_objects_similar(obj1, obj2, ignore):
     return attributes1 == attributes2
```
Comment (sebhmg) [inline geoh5py/shared/utils.py:264]: **suggestion:** in `are_objects_similar()`, have type hints, and doc for param `ignore`:
```python
def are_objects_similar(obj1, obj2, ignore: list[str]) -> bool:
```

**Review by sebhmg — APPROVED** [review]
that function looks way more readable. thank you !

If you are up to, see some suggestion for adding type hint (not blocking)

**Review by sebhmg — DISMISSED** [review]

File: geoh5py/shared/utils.py (hunk 5 above)
Comment (benk-mira) [reply geoh5py/shared/utils.py:265]: done

**Review by sebhmg — APPROVED** [review]

**Review by domfournier — APPROVED** [review]

**Pull request merged**